pkg/kubectl/resource/mapper.go

```diff
@@ -37,11 +39,19 @@ type Mapper struct {
 // if any of the decoding or client lookup steps fail. Name and namespace will be
 // set into Info if the mapping's MetadataAccessor can retrieve them.
 func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
-	json, err := yaml.ToJSON(data)
+	jsonData, err := yaml.ToJSON(data)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse %q: %v", source, err)
 	}
-	data = json
+	data = jsonData
+	
+	//To check whether the json format is correct.
+	var f interface{}
+	err = json.Unmarshal(data, &f)
```

Comment:
This isn't the right place to do this. Dig into the DataVersionAndKind method call below; it already parses w/ json package. This is a hot code path and we can't add an entirely new unmarshal call.